torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1667,6 +1667,14 @@ void ProcessGroupNCCL::heartbeatMonitor() {
         try {
           checkExceptionDump =
               globalStore_->check({std::string(EXCEPTION_DUMP)});
+        } catch (const c10::DistNetworkError& e) {
+            LOG(WARNING)
+              << logPrefix()
+              << "Failed to check the \"should dump\" flag on TCPStore, "
+              << "(maybe TCPStore server has shut down too early), with error: "
+              << e.msg();
```

Comment:
if the only difference is e.msg or e.what. shall we consolidate them?

Reply:
consolidated into a lambda